pkg/skaffold/runner/build_deploy.go

```diff
@@ -67,6 +67,19 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 
 	// Make sure all artifacts are redeployed. Not only those that were just built.
 	r.builds = build.MergeWithPreviousBuilds(bRes, r.builds)
+
+	color.Default.Fprintln(out, "Tags used in deployment:")
+
+	if r.imagesAreLocal {
+		color.Yellow.Fprintln(out, " - Since images are not pushed, they can't be referenced by digest")
+		color.Yellow.Fprintln(out, "   They are tagged and referenced by a unique ID instead")
```

Comment:
we could add more docs around this and link to those docs in this text?

Reply:
I created an issue: https://github.com/GoogleContainerTools/skaffold/issues/2795